clientd/src/lib.rs

```diff
@@ -71,38 +112,146 @@ pub async fn run_manager(
     }
 }
 
-pub fn map_subscribers_to_events() -> HashMap<EventKey, broadcast::Sender<Event>> {
-    let mut se_map = HashMap::new();
-    for event in EventKey::iter() {
-        let (tx, _rx) = broadcast::channel(128);
-        se_map.insert(event, tx);
-    }
-    se_map
-}
-
-pub async fn init_fedimint_client(cfg: PathBuf) -> Client<UserClientConfig> {
-    let cfg_path = cfg.join("client.json");
-    let db_path = cfg.join("client.db");
-    let cfg: UserClientConfig = load_from_file(&cfg_path);
-    let db = fedimint_rocksdb::RocksDb::open(db_path).unwrap().into();
-
-    Client::new(cfg.clone(), db, Default::default()).await
-}
-
-// FIXME: jsonrpsee will do the serialization for me. how can I just pass any type over the channel, the serialize -> string -> serialize feels a bit stupid
 async fn handle_rpc_request(
     request: RpcRequest,
-    fedimint_client: &mut Client<UserClientConfig>,
-) -> Result<String, RpcError> {
+    fedimint_client: Option<&FedimintClient>,
+) -> Result<HandlerResult, CallError> {
     match request {
-        RpcRequest::HealthCheck() => Ok("".to_string()),
-        RpcRequest::Info() => rpc::info(fedimint_client)
+        RpcRequest::HealthCheck() => Ok(HandlerResult::HealthCheck()),
+        RpcRequest::Info() => {
+            rpc::info(
+                fedimint_client
+                    .expect("fedimint client was passed to the dispatcher")
+                    .as_ref(),
+            )
             .await
-            .map(|info_res| serde_json::to_string(&json!(info_res)).unwrap()),
+        }
     }
 }
 
 fn load_from_file<T: DeserializeOwned>(path: &Path) -> T {
     let file = std::fs::File::open(path).expect("Can't read cfg file.");
     serde_json::from_reader(file).expect("Could not parse cfg file.")
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use std::time::Duration;
+
+    #[test]
+    fn event_subscriber_initialization_works() {
```

Comment:
This test doesn't sit with me well, because it tautologically correct (constructor uses `EventKey::iter`, so it's a bit like testing if `EventKey::iter == EventKey::iter`, and also just tests internal details and not external behavior of `EventSubscribers`

Reply:
true, I'll remove it